crates/bevy_utils/src/cow_arc.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    fmt::{Debug, Display},
+    hash::Hash,
+    ops::Deref,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+/// Much like a [`Cow`](std::borrow::Cow), but owned values are Arc-ed to make clones cheap. This should be used for values that
+/// are cloned for use across threads and change rarely (if ever).
+pub enum CowArc<'a, T: ?Sized + 'static> {
+    /// A borrowed value
+    Borrowed(&'a T),
+    /// A static value reference
+    Static(&'static T),
```

Comment:
I like the idea here since it seems like a pretty easy optimization, but I think the difference here should get a bit more documentation to explain why `Static` is needed/preferred over `Borrowed` for static references.